.gitlab-ci.yml

```diff
@@ -260,6 +274,30 @@ publish:s3:
   only:
     - /^(master|[0-9]+\.[0-9]+\.x)$/
 
+publish:docker:
+  stage: publish
+  rules:
+    - if: '$CI_COMMIT_TAG'
+    - if: '$CI_COMMIT_BRANCH == "master"'
+    - when: manual # Fallback; allow manual trigger
+      allow_failure: true
+  image:
+    name: "quay.io/skopeo/stable:v1.15"
+    # https://docs.gitlab.com/ee/ci/docker/using_docker_images.html#override-the-entrypoint-of-an-image
+    entrypoint: [""]
+  before_script:
+    - skopeo login --username "$CI_REGISTRY_USER" --password "$CI_REGISTRY_PASSWORD" "$CI_REGISTRY"
+    - |
+      skopeo login \
+        --username "$DOCKER_HUB_USERNAME" \
+        --password "$DOCKER_HUB_PASSWORD" \
+        docker.io
```

Comment:
I don't know if `skopeo login` supports also `--password-stdin`, but if so please use it here too

Reply:
It does, the CLI is pretty much identical.